### PR TITLE
docs: translate Managing State

### PR DESCRIPTION
**Pull request by andriansanto**
Closes https://github.com/reactjs/id.react.dev/issues/378

## Description

Translate the https://id.react.dev/learn/managing-state page
Page URL: https://id.react.dev/learn/managing-state

### CODE REVIEW

File: src/content/learn/managing-state.md [hunk 1]
```diff
@@ -4,27 +4,27 @@ title: Managing State
 
 <Intro>
 
-As your application grows, it helps to be more intentional about how your state is organized and how the data flows between your components. Redundant or duplicate state is a common source of bugs. In this chapter, you'll learn how to structure your state well, how to keep your state update logic maintainable, and how to share state between distant components.
+Ketika aplikasi Anda berkembang, penting untuk lebih memperhatikan bagaimana keadaan (*state*) Anda diorganisasi dan bagaimana aliran data antara komponen-komponen Anda. Keadaan yang redundan atau duplikat adalah sumber umum dari bug. Di bab ini, Anda akan belajar bagaimana mengatur keadaan (*state*) dengan baik, bagaimana menjaga logika pembaruan keadaan (*state*) agar mudah dikelola, dan bagaimana berbagi keadaan (*state*) antara komponen-komponen yang jauh.
```
Comment (zainfathoni) [inline src/content/learn/managing-state.md:7]:
```suggestion
Ketika aplikasi Anda berkembang, penting untuk lebih memperhatikan bagaimana *state* Anda diorganisasi dan bagaimana aliran data antara komponen-komponen Anda. Keadaan yang redundan atau duplikat adalah sumber umum dari bug. Di bab ini, Anda akan belajar bagaimana mengatur *state* dengan baik, bagaimana menjaga logika pembaruan *state* agar mudah dikelola, dan bagaimana berbagi *state* antara komponen-komponen yang jauh.
```

According to the [glossary](https://github.com/reactjs/id.react.dev/wiki/Glosarium#s), we should NOT translate `state` to `keadaan`. The reason being `state` is a literal syntax actively used in React.
Therefore, please update all the "keadaan (*state*)" occurrences down below into "*state*". Thanks!

File: src/content/learn/managing-state.md [hunk 2]
```diff
@@ -693,15 +693,15 @@ ul, li { margin: 0; padding: 0; }
...
 
 </LearnMore>
 
-## Passing data deeply with context {/*passing-data-deeply-with-context*/}
+## Melewatkan data secara dalam dengan *context* {/*passing-data-deeply-with-context*/}
 
-Usually, you will pass information from a parent component to a child component via props. But passing props can become inconvenient if you need to pass some prop through many components, or if many components need the same information. Context lets the parent component make some information available to any component in the tree below it—no matter how deep it is—without passing it explicitly through props.
+Biasanya, Anda akan melewatkan informasi dari komponen induk (*parent*) ke komponen anak (*children*) melalui `props`. Namun, melewatkan `props` dapat menjadi merepotkan jika Anda perlu melewatkan beberapa `prop` melalui banyak komponen, atau jika banyak komponen membutuhkan informasi yang sama. *Context* memungkinkan komponen induk membuat beberapa informasi tersedia untuk setiap komponen di bawahnya—tidak peduli seberapa dalam itu—tanpa melewatkan secara eksplisit melalui `props`.
```
Comment (zainfathoni) [inline src/content/learn/managing-state.md:702]:
```suggestion
Biasanya, Anda akan melewatkan informasi dari komponen induk ke komponen anak (*children*) melalui `props`. Namun, melewatkan `props` dapat menjadi merepotkan jika Anda perlu melewatkan beberapa `prop` melalui banyak komponen, atau jika banyak komponen membutuhkan informasi yang sama. *Context* memungkinkan komponen induk membuat beberapa informasi tersedia untuk setiap komponen di bawahnya—tidak peduli seberapa dalam itu—tanpa melewatkan secara eksplisit melalui `props`.
```

According to [this writing guide](https://github.com/reactjs/id.react.dev/wiki/Panduan-Penulisan-Universal#2-memperkenalkan-kata-baru-dalam-bahasa-indonesia-sambil-mengutip-kata-aslinya), once you introduced "induk (*parent*)" translation for the first time above (at line `234`), you don't need to repeat it for the subsequent occurrences. So you may leave the "(*parent*)" part out.

image

File: src/content/learn/managing-state.md [hunk 3]
```diff
@@ -1006,12 +1006,12 @@ ul, li { margin: 0; padding: 0; }
 
 <LearnMore path="/learn/scaling-up-with-reducer-and-context">
 
-Read **[Scaling Up with Reducer and Context](/learn/scaling-up-with-reducer-and-context)** to learn how state management scales in a growing app.
+Baca **[Mengembangkan Skala dengan *Reducer* dan *Context*](/learn/scaling-up-with-reducer-and-context)** untuk mempelajari bagaimana pengelolaan keadaan (*state*) mengembang pada aplikasi yang berkembang.
```
Comment (zainfathoni) [inline src/content/learn/managing-state.md:1009]:
```suggestion
Baca **[Peningkatan Skala dengan *Reducer* dan *Context*](/learn/scaling-up-with-reducer-and-context)** untuk mempelajari bagaimana pengelolaan keadaan (*state*) mengembang pada aplikasi yang berkembang.
```

Please translate links' title according to the [translated title](https://id.react.dev/learn/scaling-up-with-reducer-and-context).

**Review by zainfathoni — CHANGES_REQUESTED** [review]
Thanks for contributing, @andriansanto! 🙏
Though I have a few comments that might lead to significant changes.
So please address them first, and then I will give it another review once you addressed them all. 😁

**Comment by andriansanto** [comment]
> Thanks for contributing, @andriansanto! 🙏 Though I have a few comments that might lead to significant changes. So please address them first, and then I will give it another review once you addressed them all. 😁

Thankyou for the feedback, will solve it soon 😄

**Comment by andriansanto** [comment]
> > Thanks for contributing, @andriansanto! 🙏 Though I have a few comments that might lead to significant changes. So please address them first, and then I will give it another review once you addressed them all. 😁
> 
> Thankyou for the feedback, will solve it soon 😄

Done, Kindly for the review again @zainfathoni  🙏

**Review by resir014 — CHANGES_REQUESTED** [review]
Don't forget to translate the page title as well.

```md
---
title: Mengelola State
---
```

File: src/content/learn/managing-state.md [hunk 4]
```diff
@@ -1,30 +1,30 @@
 ---
-title: Managing State
+title: Mengelola State
 ---
 
 <Intro>
 
-As your application grows, it helps to be more intentional about how your state is organized and how the data flows between your components. Redundant or duplicate state is a common source of bugs. In this chapter, you'll learn how to structure your state well, how to keep your state update logic maintainable, and how to share state between distant components.
+Ketika aplikasi Anda berkembang, penting untuk lebih memperhatikan bagaimana *state* Anda diorganisasi dan bagaimana aliran data antara komponen-komponen Anda. *State* yang redundan atau duplikat adalah sumber umum dari bug. Di bab ini, Anda akan belajar bagaimana mengatur *state* dengan baik, bagaimana menjaga logika pembaruan *state* agar mudah dikelola, dan bagaimana berbagi *state* antara komponen-komponen yang jauh.
```
Comment (r17x) [inline src/content/learn/managing-state.md:7]: `As your application grows` have been discussed in https://github.com/reactjs/id.react.dev/pull/351#discussion_r1179213235 , should we make it same as [`useContext` translation](https://github.com/reactjs/id.react.dev/blob/main/src/content/reference/react/useContext.md?plain=1#L568) or another? @resir014 @mhaidarhanif @zainfathoni .

Comment (zainfathoni) [reply src/content/learn/managing-state.md:7]: Yes, we need to do it consistently.

The decision for https://github.com/reactjs/id.react.dev/pull/351#discussion_r1179213235 was `Seiring dengan bertumbuhnya aplikasi Anda`, right?

Comment (zainfathoni) [reply src/content/learn/managing-state.md:7]:
```suggestion
Seiring dengan bertumbuhnya aplikasi Anda, penting untuk lebih memperhatikan bagaimana *state* Anda diorganisasi dan bagaimana data mengalir di antara komponen-komponen Anda. *State* yang redundan atau duplikat adalah sumber umum dari *bug*. Di bab ini, Anda akan belajar bagaimana mengatur *state* dengan baik, bagaimana menjaga logika pembaruan *state* agar mudah dikelola, dan bagaimana berbagi *state* antara komponen-komponen yang jauh.
```

I just suggested it here, https://github.com/reactjs/id.react.dev/discussions/337#discussioncomment-5772140

**Comment by mhaidarhanif** [comment]
@resir014 and @zainfathoni can check and approve if okay.

**Review by resir014 — APPROVED** [review]
OK to go with @huedaya's additions. Will cherry-pick and merge when I have the time.

**Pull request merged**